src/simsopt/field/coil.py

```diff
@@ -172,36 +299,133 @@ def apply_symmetries_to_currents(base_currents, nfp, stellsym):
+        net_torques[i, :] = np.array([coil_net_torque(c, coils)])
+        if close:
+            coil_force_temp = np.vstack((coil_force_temp, coil_force_temp[0, :]))
+            coil_torque_temp = np.vstack((coil_torque_temp, coil_torque_temp[0, :]))
+        coil_forces[i * ppl[i]: (i + 1) * ppl[i], :] = coil_force_temp
+        coil_torques[i * ppl[i]: (i + 1) * ppl[i], :] = coil_torque_temp
+
+    coil_data = np.zeros((data.shape[0], 3))
+    for i in range(len(coils)):
+        coil_data[i * ppl[i]: (i + 1) * ppl[i], :] = net_forces[i, :]
+    coil_data = np.ascontiguousarray(coil_data)
+    pointData['NetForces'] = (contig(coil_data[:, 0]),
+                                contig(coil_data[:, 1]),
+                                contig(coil_data[:, 2]))
+    coil_data = np.zeros((data.shape[0], 3))
+    for i in range(len(coils)):
+        coil_data[i * ppl[i]: (i + 1) * ppl[i], :] = net_torques[i, :]
+    coil_data = np.ascontiguousarray(coil_data)
+    pointData['NetTorques'] = (contig(coil_data[:, 0]),
+                                contig(coil_data[:, 1]),
+                                contig(coil_data[:, 2]))
+    pointData["Pointwise_Forces"] = (contig(coil_forces[:, 0]), contig(coil_forces[:, 1]), contig(coil_forces[:, 2]))
+    pointData["Pointwise_Torques"] = (contig(coil_torques[:, 0]), contig(coil_torques[:, 1]), contig(coil_torques[:, 2]))
+    if extra_data is not None:
+        pointData = {**pointData, **extra_data}
+    curves_to_vtk(curves, filename, close=close, extra_data=pointData)
+
+def coils_via_symmetries(curves, currents, nfp, stellsym, regularizations=None):
+    """
+    Generate a list of Coil objects by applying rotational and (optionally) stellarator symmetries.
+
     Take a list of ``n`` curves and return ``n * nfp * (1+int(stellsym))``
     ``Coil`` objects obtained by applying rotations and flipping corresponding
     to ``nfp`` fold rotational symmetry and optionally stellarator symmetry.
+
+    Parameters
+    ----------
+    curves : list of Curve
+        List of base curves.
+    currents : list of Current
+        List of base current objects.
+    nfp : int
+        Number of field periods (rotational symmetry).
+    stellsym : bool
+        Whether to apply stellarator symmetry.
+    regularization : Regularization
+        The regularization object for the coils representing the finite coil cross section.
+        Default is a circular cross section with radius 0.05.
+
+    Returns
+    -------
+    coils : list of Coil
+        List of Coil objects with symmetries applied.
     """
 
     assert len(curves) == len(currents)
     curves = apply_symmetries_to_curves(curves, nfp, stellsym)
     currents = apply_symmetries_to_currents(currents, nfp, stellsym)
-    coils = [Coil(curv, curr) for (curv, curr) in zip(curves, currents)]
+    if regularizations is None:
+        regularizations = [regularization_circ(0.05) for _ in curves]
```

Comment:
I'm not sure about including a regularization in a coil when some problems don't have this, I would have thought this would default to `None`